tests/framework/e2e/cluster.go

```diff
@@ -457,22 +458,24 @@ func (cfg *EtcdProcessClusterConfig) SetInitialOrDiscovery(serverCfg *EtcdServer
 
 func (cfg *EtcdProcessClusterConfig) EtcdServerProcessConfig(tb testing.TB, i int) *EtcdServerProcessConfig {
 	var curls []string
-	var curl, curltls string
+	var curl string
 	port := cfg.BasePort + 5*i
 	clientPort := port
 	peerPort := port + 1
 	metricsPort := port + 2
 	peer2Port := port + 3
+	clientHttpPort := port + 4
 
-	curlHost := fmt.Sprintf("localhost:%d", clientPort)
-	switch cfg.Client.ConnectionType {
-	case ClientNonTLS, ClientTLS:
-		curl = (&url.URL{Scheme: cfg.ClientScheme(), Host: curlHost}).String()
+	if cfg.Client.ConnectionType == ClientTLSAndNonTLS {
+		curl = clientURL(clientPort, ClientNonTLS)
+		curls = []string{curl, clientURL(clientPort, ClientTLS)}
+	} else {
+		curl = clientURL(clientPort, cfg.Client.ConnectionType)
 		curls = []string{curl}
-	case ClientTLSAndNonTLS:
-		curl = (&url.URL{Scheme: "http", Host: curlHost}).String()
-		curltls = (&url.URL{Scheme: "https", Host: curlHost}).String()
-		curls = []string{curl, curltls}
+	}
+	clientHttpUrl := curl
```

Comment:
`var clientHttpUrl string`. I think the initial value should be empty.

Reply:
Done